### PR TITLE
tests: secret controller

### LINKED ISSUE #65: Add secrets controller unit tests

### PR DESCRIPTION
**Pull request by dvirgilad**
With this pr, tests have been added for the secrets controller
fixes #65

### CODE REVIEW

**Review by mzeevi — CHANGES_REQUESTED** [review]
Needs some changing to align with the latest changes

File: src/controllers/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+			if test.want.errorStatus != v1.StatusSuccess {
+				reason := err.(errors.APIStatus).Status().Reason
+
+				assert.Equal(t, test.want.errorStatus, reason)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, test.want.response, response)
+		})
+
```
Comment (fran2810) [inline src/controllers/secret_test.go:98]: remove redundant spaces

File: src/controllers/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+	createTestSecret(secretName+"2", namespaceName)
+	createTestNamespace("NoSecretsNamespace")
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			response, err := secretController.GetSecrets(test.requestParams.namespace)
+			assert.NoError(t, err)
+
+			assert.Equal(t, test.want.response, response)
+		})
+
```
Comment (fran2810) [inline src/controllers/secret_test.go:164]: what is this

**Review by mzeevi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**